### PR TITLE
[KEYCLOAK-4932] - Spring Boot quickstart using fine-grained permissions

### PR DESCRIPTION
**Pull request by pedroigor**
Depends on https://github.com/keycloak/keycloak/pull/4156

### CODE REVIEW

File: app-authz-springboot/src/test/resources/application.properties [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+server.port = 8080
+keycloak.realm=spring-boot-quickstart
+keycloak.auth-server-url=http://localhost:8180/auth
+keycloak.ssl-required=external
+keycloak.resource=app-authz-springboot
+keycloak.public-client=false
+keycloak.credentials.secret=secret
+product.service.url=http://localhost:8081/products
+keycloak.policy-enforcer-config.on-deny-redirect-to=/accessDenied
+keycloak.securityConstraints[1].authRoles[0] = user
```
Comment (sebastienblanc) [inline app-authz-springboot/src/test/resources/application.properties]: Not a big deal but it should be `securityConstraints[0]` instead of `[1]`

**Comment by sebastienblanc** [comment]
@pedroigor Could you rebase this PR ? We just merged some changes that moves some classes around ;)

**Comment by pedroigor** [comment]
@sebastienblanc Sure. But after the merge the quickstart is not working anymore. Even tests are broken. Will see what I need to change.

**Comment by pedroigor** [comment]
@sebastienblanc rebased. I've also added some more docs and removed unnecessary code.

Tks.

**Comment by abstractj** [comment]
@sebastienblanc ^^^

**Comment by sebastienblanc** [comment]
:eyes:

**Pull request merged**